pkg/postage/batchstore/store.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package batchstore
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/storage"
+)
+
+var (
+	batchKeyPrefix = "batchKeyPrefix"
+	valueKeyPrefix = "valueKeyPrefix"
+)
+
+var _ postage.EventUpdater = (*Store)(nil)
+
+// Store is a local store  for postage batches
+type Store struct {
+	store  storage.StateStorer // state store backend to persist batches
+	mu     sync.Mutex          // mutex to lock statestore during atomic changes
+	state  *state              // the current state
+	logger logging.Logger
+}
+
+// New constructs a new postage batch store
+func New(store storage.StateStorer, logger logging.Logger) (*Store, error) {
+	// initialise state from statestore or start with 0-s
+	st := &state{}
+	if err := st.load(store); err != nil {
+		return nil, err
+	}
+	s := &Store{
+		store:  store,
+		logger: logger,
+	}
+	return s, nil
+}
+
+// settle retrieves the current state
+// - sets the cumulative outpayment normalised, cno+=price*period
+// - sets the new block number
+// caller holds the store mutex
+func (s *Store) settle(block uint64) {
+	period := int64(block - s.state.block)
+	s.state.block = block
+	s.state.total.Add(s.state.total, new(big.Int).Mul(s.state.price, big.NewInt(period)))
```

Comment:
this reintroduces a lot of the reorg issues we wanted to get rid of. instead of doing this price update thing here we should refetch this value using the `totalOutPayment` getter when there is a `PriceUpdate` event on the oracle contract.